### PR TITLE
(BKR-1076) Refactor beaker init and provision

### PR DESCRIPTION
**Pull request by tvpartytonight**
This changes beaker init to focus on setting up a repository for
beaker subcommands, rather than focusing in on generating host files
themselves. This also adds the functionality to create files that
contain the configured options; that is, all options that are not
presets after beaker parses its OptionsHash.

### CODE REVIEW

**Comment by tvpartytonight** [comment]
@puppetlabs-jenkins test this please

**Comment by tvpartytonight** [comment]
@puppetlabs-jenkins retest this please

**Comment by tvpartytonight** [comment]
@puppetlabs-jenkins retest this please

**Comment by tvpartytonight** [comment]
@puppetlabs-jenkins retest this please

File: acceptance/tests/subcommands/init.rb [hunk 1]
```diff
@@ -4,45 +4,20 @@ def delete_root_folder_contents
...
-      assert_match(/Writing host config.+/, result.stdout)
-      step 'ensure that the Rakefile is present' do
-        on(default, '[ -e "Rakefile" ]')
-      end
+  step 'ensure beaker init writes YAML configuration files to disk' do
     delete_root_folder_contents
-    end
+    on(default, 'beaker init')
+    subcommand_options = on(default, 'cat .beaker/subcommand_options.yaml').stdout
+    subcommand_state = on(default, 'cat .beaker/.subcommand_state.yaml').stdout
```
Comment (kevpl) [inline acceptance/tests/subcommands/init.rb]: @tvpartytonight can we not use the constants for the paths here?

**Comment by kevpl** [comment]
other than the above comment, which might have been answered previously, I'm 👍  on this

**Comment by ferglor** [comment]
👍

File: acceptance/tests/subcommands/init.rb (hunk 1 above)
Comment (tvpartytonight) [inline acceptance/tests/subcommands/init.rb]: We can use the constants here 👍

**Comment by tvpartytonight** [comment]
@puppetlabs-jenkins retest this please

**Pull request merged**